dart-impl/base/include/dash/dart/base/mutex.h

```diff
@@ -111,7 +111,7 @@ dart__base__mutex_trylock(dart_mutex_t *mutex)
 #ifdef DART_HAVE_PTHREADS
   int ret = pthread_mutex_trylock(&mutex->mutex);
   DART_LOG_TRACE("dart__base__mutex_trylock: lock %p aqcuired: %s",
-                 mutex, (ret == 0) ? "yes" : "no");
+                 (const void *)mutex, (ret == 0) ? "yes" : "no");
```

Comment:
Why do we need the explicit cast here (and elsewhere)?

Reply:
The standard says (§ 7.19.6.1):
`p`  The argument shall be a pointer to void. The value of the pointer is converted to a sequence of printing characters, in an implementation-defined manner.

Reply:
The compilers prefers to have it explicit.

Reply:
This seems overly strict to me and makes printing a pointer `-Wall`-clean ugly and cumbersome. Implicit casts to and from `void*` are perfectly valid in C, e.g., `malloc` and `free`, and afaics it does not make any difference for `printf`'s `%p` what type the pointer has. Are all compilers warning about this? 

IMO, unless there is a really good reason for enforcing these casts we should rather suppress them.

Reply:
We can suppress this warning with the proper compiler flag. Then we can remove all these casts.

Reply:
:+1: